### ISSUE TITLE
Plugin 2.0.9 causes infinite re-indexing

### ISSUE THREAD

**Issue by iRevive** [iancaffey/smithy-intellij-plugin#10]
OS: MacOS Ventura 13.4
Intellij IDEA:  2023.1.2, Build #IU-231.9011.34
Smithy plugin: 2.0.9

When I updated Smithy plugin to 2.0.9, IntelliJ IDEA got stuck in the infinite 'update indexes' loop. Once I disabled the plugin, the infinite re-indexing stopped. 
The project does not even have Smithy files.

Here is the log:
```
2023-06-02 11:59:48,772 [348707460]   INFO - #c.i.u.i.UnindexedFilesIndexer - Use 7 indexing threads for indexing of [project_name_reducted]
2023-06-02 11:59:48,814 [348707502]   INFO - #c.i.u.i.i.MapReduceIndex - An exception during updateWithMap(). Index smithy.ast-shapes will be rebuilt.
com.intellij.util.indexing.StorageException: com.intellij.util.indexing.StorageException: java.io.UTFDataFormatException: encoded string ({"type":..."1.0"}}}) too long: 209869 bytes
	at com.intellij.util.indexing.impl.MapReduceIndex.updateWithMap(MapReduceIndex.java:359)
	at com.intellij.util.indexing.storage.MapReduceIndexBase.updateWithMap(MapReduceIndexBase.java:105)
	at com.intellij.util.indexing.impl.MapReduceIndex$IndexUpdateComputable.compute(MapReduceIndex.java:388)
	at com.intellij.util.indexing.impl.MapReduceIndex$IndexUpdateComputable.compute(MapReduceIndex.java:370)
	at com.intellij.openapi.util.Computable.get(Computable.java:16)
	at com.intellij.indexing.composite.CompositeInvertedIndexBase.lambda$mapInputAndPrepareUpdate$3(CompositeInvertedIndexBase.java:77)
	at com.intellij.util.indexing.IndexInfrastructureExtensionUpdateComputation.compute(IndexInfrastructureExtensionUpdateComputation.java:27)
	at com.intellij.util.indexing.IndexInfrastructureExtensionUpdateComputation.compute(IndexInfrastructureExtensionUpdateComputation.java:11)
	at com.intellij.openapi.util.Computable.get(Computable.java:16)
	at com.intellij.util.indexing.FileBasedIndexImpl.lambda$runUpdateForPersistentData$27(FileBasedIndexImpl.java:1762)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:679)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computeUnderProgress(CoreProgressManager.java:635)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$computeInNonCancelableSection$4(CoreProgressManager.java:230)
	at com.intellij.openapi.progress.Cancellation.computeInNonCancelableSection(Cancellation.java:99)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computeInNonCancelableSection(CoreProgressManager.java:230)
	at com.intellij.util.indexing.FileBasedIndexImpl.lambda$runUpdateForPersistentData$28(FileBasedIndexImpl.java:1762)
	at com.intellij.util.indexing.StorageBufferingHandler.runUpdate(StorageBufferingHandler.java:22)
	at com.intellij.util.indexing.FileBasedIndexImpl.runUpdateForPersistentData(FileBasedIndexImpl.java:1761)
	at com.intellij.util.indexing.SingleIndexValueApplier.doApply(SingleIndexValueApplier.java:65)
	at com.intellij.util.indexing.SingleIndexValueApplier.apply(SingleIndexValueApplier.java:53)
	at com.intellij.util.indexing.FileIndexesValuesApplier.apply(FileIndexesValuesApplier.java:111)
	at com.intellij.util.indexing.contentQueue.IndexUpdateRunner.doWriteIndexesForFile(IndexUpdateRunner.java:365)
	at com.intellij.util.indexing.contentQueue.IndexUpdateRunner.writeIndexesForFile(IndexUpdateRunner.java:353)
	at com.intellij.util.indexing.contentQueue.IndexUpdateRunner.indexOneFileOfJob(IndexUpdateRunner.java:327)
	at com.intellij.util.indexing.contentQueue.IndexUpdateRunner.lambda$indexJobsFairly$3(IndexUpdateRunner.java:243)
	at com.intellij.openapi.progress.impl.ProgressSuspender.executeNonSuspendableSection(ProgressSuspender.java:80)
	at com.intellij.util.indexing.contentQueue.IndexUpdateRunner.indexJobsFairly(IndexUpdateRunner.java:246)
	at com.intellij.util.indexing.contentQueue.IndexUpdateRunner.lambda$doIndexFiles$2(IndexUpdateRunner.java:177)
	at com.intellij.util.concurrency.BoundedTaskExecutor.doRun(BoundedTaskExecutor.java:248)
	at com.intellij.util.concurrency.BoundedTaskExecutor.access$200(BoundedTaskExecutor.java:28)
	at com.intellij.util.concurrency.BoundedTaskExecutor$1.executeFirstTaskAndHelpQueue(BoundedTaskExecutor.java:221)
	at com.intellij.util.ConcurrencyUtil.runUnderThreadName(ConcurrencyUtil.java:227)
	at com.intellij.util.concurrency.BoundedTaskExecutor$1.run(BoundedTaskExecutor.java:210)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:702)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:699)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1.run(Executors.java:699)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: com.intellij.util.indexing.StorageException: java.io.UTFDataFormatException: encoded string ({"type":..."1.0"}}}) too long: 209869 bytes
	at com.intellij.util.indexing.impl.MapIndexStorage.unwrapCauseAndRethrow(MapIndexStorage.java:327)
	at com.intellij.util.indexing.impl.MapIndexStorage.read(MapIndexStorage.java:259)
	at com.intellij.util.indexing.impl.MapIndexStorage.addValue(MapIndexStorage.java:170)
	at com.intellij.util.indexing.impl.storage.VfsAwareMapIndexStorage.addValue(VfsAwareMapIndexStorage.java:139)
	at com.intellij.util.indexing.impl.storage.TransientChangesIndexStorage.addValue(TransientChangesIndexStorage.java:153)
	at com.intellij.util.indexing.impl.MapReduceIndex$2.process(MapReduceIndex.java:329)
	at com.intellij.util.indexing.impl.EmptyInputDataDiffBuilder.processAllKeyValuesAsAdded(EmptyInputDataDiffBuilder.java:48)
	at com.intellij.util.indexing.impl.MapInputDataDiffBuilder.differentiate(MapInputDataDiffBuilder.java:44)
	at com.intellij.util.indexing.impl.UpdateData.iterateKeys(UpdateData.java:57)
	at com.intellij.util.indexing.impl.MapReduceIndex.updateWithMap(MapReduceIndex.java:351)
	... 39 more
Caused by: java.io.UTFDataFormatException: encoded string ({"type":..."1.0"}}}) too long: 209869 bytes
	at java.base/java.io.DataOutputStream.writeUTF(DataOutputStream.java:369)
	at java.base/java.io.DataOutputStream.writeUTF(DataOutputStream.java:333)
	at software.amazon.smithy.intellij.index.SmithyAstShapeIndex$getValueExternalizer$1.save(SmithyAstShapeIndex.kt:43)
	at software.amazon.smithy.intellij.index.SmithyAstShapeIndex$getValueExternalizer$1.save(SmithyAstShapeIndex.kt:40)
	at com.intellij.util.indexing.impl.ValueContainerImpl.saveTo(ValueContainerImpl.java:451)
	at com.intellij.util.indexing.impl.ChangeTrackingValueContainer.saveTo(ChangeTrackingValueContainer.java:186)
	at com.intellij.util.indexing.impl.ValueContainerMap$1.append(ValueContainerMap.java:54)
	at com.intellij.util.io.PersistentMapImpl.doAppendData(PersistentMapImpl.java:543)
	at com.intellij.util.io.PersistentMapImpl.appendData(PersistentMapImpl.java:513)
	at com.intellij.util.indexing.impl.ValueContainerMap.merge(ValueContainerMap.java:51)
	at com.intellij.util.indexing.impl.MapIndexStorage.onDropFromCache(MapIndexStorage.java:94)
	at com.intellij.util.indexing.impl.MapIndexStorage.lambda$initMapAndCache$1(MapIndexStorage.java:75)
	at com.intellij.util.indexing.impl.MapIndexStorageSlruCache$cache$1.onDropFromCache(MapIndexStorageCache.kt:64)
	at com.intellij.util.indexing.impl.MapIndexStorageSlruCache$cache$1.onDropFromCache(MapIndexStorageCache.kt:58)
	at com.intellij.util.containers.SLRUMap$2.removeEldestEntry(SLRUMap.java:56)
	at com.intellij.util.containers.hash.LinkedHashMap.put(LinkedHashMap.java:112)
	at com.intellij.util.containers.SLRUMap.put(SLRUMap.java:93)
	at com.intellij.util.containers.SLRUCache.get(SLRUCache.java:34)
	at com.intellij.util.indexing.impl.MapIndexStorageSlruCache.read(MapIndexStorageCache.kt:69)
	at com.intellij.util.indexing.impl.MapIndexStorage.read(MapIndexStorage.java:256)
	... 47 more
```

**Comment by iancaffey** [comment]
Thanks for the report, Maksym.

Looks like your project has a dependency with a single Smithy shape which serializes to JSON having a size of 209,869 bytes.

https://smithy.io/2.0/spec/json-ast.html

---

Would you mind pulling the plugin jar from [v2.0.10](https://github.com/iancaffey/smithy-intellij-plugin/releases/tag/v2.0.10), installing it locally from disk into your IntelliJ, and reporting back if it resolves your indexing problem?

v2.0.10 moves off of that API in DataOuputStream which has an implicit 65kb size limit.

**Comment by iRevive** [comment]
Thanks for the assistance! I have zero issues with v2.0.10. The project loads fine, no re-indexing was spotted.

**Comment by iancaffey** [comment]
Great to hear! 

I've released v2.0.10, should be available in < 24 hours, once JetBrains approves the update.
